tests/src/test/java/alluxio/master/JournalShutdownIntegrationTest.java

```diff
@@ -90,28 +93,63 @@ public final void before() throws Exception {
 
   @Test
   public void singleMasterJournalStopIntegration() throws Exception {
-    LocalAlluxioCluster cluster = setupSingleMasterCluster();
-    runCreateFileThread(cluster.getClient());
-    System.out.println(Configuration.get(PropertyKey.MASTER_JOURNAL_FOLDER));
-    // Shutdown the cluster
-    cluster.stopFS();
-    CommonUtils.sleepMs(TEST_TIME_MS);
-    awaitClientTermination();
-    reproduceAndCheckState(mCreateFileThread.getSuccessNum());
+    ExternalCluster cluster = ExternalCluster.newBuilder()
+        .setClusterName("singleMasterJournalStopIntegration")
+        .setNumWorkers(0)
+        .setNumMasters(1)
+        .build();
+    try {
+      cluster.start();
+      FileSystem fs = cluster.getFileSystemClient();
+      runCreateFileThread(fs);
+      cluster.killPrimaryMaster();
+      cluster.startMaster(0);
```

Comment:
Should we have a different API for just bringing up the cluster afterward?

Reply:
To keep things simple, the cluster doesn't keep track of which processes are running or destroyed - this management is left to individual tests. We could have add a "startAll" method which loops starts all masters and workers, but I'd prefer to be explicit in tests about what is being restarted.

Reply:
We provide the API `killPrimaryMaster`, and the caller has no idea which master is killed, should that return the master index?

Reply:
Good idea, added